### ISSUE TITLE
[BUG] `XGBoost` evaluation is calculated against its predictions and not ground truth resulting in a perfect score

### ISSUE THREAD

**Issue by radekosmulski** [NVIDIA-Merlin/models#598]
### Bug description

In `merlin/models/xgb/__init__.py` on line `188` we are creating a `DMatrix` to be used for evaluation, but are setting `label` to the predictions of the model instead of to ground truth.

Subsequently, we pass the `DMatrix` to `self.booster.eval(eval_data)` which predicts on the passed data and calculates metrics using predictions and `label` stored in the `DMatrix`.

![image](https://user-images.githubusercontent.com/2444926/180157332-cca2ebca-6af6-4f3d-b944-0258ba3d903b.png)

### Steps/Code to reproduce bug
![image](https://user-images.githubusercontent.com/2444926/180157521-7a8631d0-d675-4cc4-b3ee-661508642390.png)

```
from merlin.core.utils import Distributed
from merlin.models.xgb import XGBoost
import sklearn.datasets
import cudf
import numpy as np
import nvtabular as nvt

rows = 200
num_features = 16
X, y = sklearn.datasets.make_regression(
    n_samples=rows,
    n_features=num_features,
    n_informative=num_features // 3,
    random_state=0,
)

feature_names = [str(i) for i in range(num_features)]
df = cudf.DataFrame(np.hstack((X, y.reshape(-1, 1))), columns=feature_names+['target'], dtype=np.float32)
train_ds = nvt.Dataset(df.iloc[:160])
valid_ds = nvt.Dataset(df.iloc[40:])

with Distributed():
    model = XGBoost(schema=train_ds.schema, target_columns='target')
    model.fit(
        train_ds,
        evals=[(valid_ds, 'validation_set'),]
    )

with Distributed():
    out = model.evaluate(valid_ds)

out
```

### Expected behavior
Metrics are calculated against ground truth and not model predictions

**Comment by viswa-nvidia** [comment]
@radekosmulski , what is the severity of this bug? Does this qualify for a P0 ?

**Comment by karlhigley** [comment]
Seems P0-ish to me. Put together a fix for it.
